### ISSUE TITLE
There is no L1 weightDecay in Torch.

### ISSUE THREAD

**Issue by amirkhango** [torch/optim#120]
The weightDecay in Torch refers to L2, Yes?
So How can I implemment L1 weightDecay?

**Comment by karandwivedi42** [comment]
Does this work? https://github.com/torch/nn/blob/master/doc/criterion.md#nn.AbsCriterion

**Comment by Kaixhin** [comment]
Example code for L1 and L2 weight decay (which is added manually in the training loop) can be found [here](https://github.com/torch/demos/blob/master/train-a-digit-classifier/train-on-mnist.lua#L214).
